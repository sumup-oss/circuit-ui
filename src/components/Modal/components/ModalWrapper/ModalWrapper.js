/**
 * Copyright 2019, SumUp Ltd.
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React from 'react';
import PropTypes from 'prop-types';
import styled from '@emotion/styled';
import { css } from '@emotion/core';

import Card from '../../../Card';

const baseStyles = ({ theme }) => css`
  width: 100%;

  ${theme.mq.untilKilo} {
    border-bottom-left-radius: 0;
    border-bottom-right-radius: 0;
    min-width: initial;
    position: relative;
  }
`;

const Wrapper = styled(Card)`
  ${baseStyles};
`;

Wrapper.defaultProps = Card.defaultProps;

const ModalWrapper = ({ ...props }) => (
  <Wrapper shadow={Card.TRIPLE} {...props} />
);

ModalWrapper.propTypes = {
  /*
   * Modal content
   */
  children: PropTypes.node.isRequired
};

/**
 * @component
 */
export default ModalWrapper;
