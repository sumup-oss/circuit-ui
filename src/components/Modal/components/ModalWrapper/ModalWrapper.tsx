/**
 * Copyright 2019, SumUp Ltd.
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React, { FC } from 'react';
import { css } from '@emotion/core';

import styled, { StyleProps } from '../../../../styles/styled';
import Card from '../../../Card';

// TODO: Extend CardProps once the Card has been migrated to TypeScript.
export interface ModalWrapperProps {
  /**
   * The shadow depth of the Modal.
   */
  shadow?: 'single' | 'double' | 'triple';
  /**
   * The padding of the Modal.
   */
  spacing?: 'mega' | 'giga';
}

const baseStyles = ({ theme }: StyleProps) => css`
  width: 100%;

  ${theme.mq.untilKilo} {
    border-bottom-left-radius: 0;
    border-bottom-right-radius: 0;
    min-width: initial;
    position: relative;
  }
`;

// FIXME: Remove any typecast once the Card has been migrated to TypeScript.
const Wrapper = styled(Card as any)<ModalWrapperProps>(baseStyles);

export const ModalWrapper: FC<ModalWrapperProps> = props => (
  <Wrapper shadow="triple" spacing="giga" {...props} />
);
