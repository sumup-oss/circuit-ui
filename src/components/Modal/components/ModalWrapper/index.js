import ModalWrapper from './ModalWrapper';

export default ModalWrapper;
