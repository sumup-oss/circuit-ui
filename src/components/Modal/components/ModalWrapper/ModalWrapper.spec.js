import React from 'react';

import ModalWrapper from '.';

describe('ModalWrapper', () => {
  /**
   * Style tests.
   */
  it('should render with default styles', () => {
    const actual = create(<ModalWrapper />);
    expect(actual).toMatchSnapshot();
  });

  /**
   * Accessibility tests.
   */
  it('should meet accessibility guidelines', async () => {
    const wrapper = renderToHtml(<ModalWrapper />);
    const actual = await axe(wrapper);
    expect(actual).toHaveNoViolations();
  });
});
