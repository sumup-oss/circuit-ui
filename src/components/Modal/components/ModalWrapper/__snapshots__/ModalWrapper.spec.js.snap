// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalWrapper should render with default styles 1`] = `
.circuit-0 {
  background-color: #FFFFFF;
  border-radius: 4px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  box-shadow: 0 0 0 1px rgba(12,15,20,0.07), 0 4px 4px 0 rgba(12,15,20,0.07),0 8px 8px 0 rgba(12,15,20,0.07);
  padding: 16px 24px;
  width: 100%;
}

@media (max-width:479px) {
  .circuit-0 {
    border-bottom-left-radius: 0;
    border-bottom-right-radius: 0;
    min-width: initial;
    position: relative;
  }
}

<div
  class="circuit-0 circuit-1"
  spacing="giga"
/>
`;
